jobs/execute_with_monitoring.go

```diff
@@ -0,0 +1,72 @@
+package jobs
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"github.com/checkmarble/marble-backend/tracing"
+	"github.com/checkmarble/marble-backend/usecases"
+	"github.com/checkmarble/marble-backend/utils"
+	"github.com/getsentry/sentry-go"
+)
+
+func executeWithMonitoring(
+	ctx context.Context,
+	uc usecases.Usecases,
+	config tracing.Configuration,
+	jobName string,
+	fn func(context.Context, usecases.Usecases) error,
+) error {
+	telemetryRessources, err := tracing.Init(config)
+	if err != nil {
+		return fmt.Errorf("error initializing tracing: %w", err)
+	}
+	ctx = utils.StoreOpenTelemetryTracerInContext(ctx, telemetryRessources.Tracer)
+
+	logger := utils.LoggerFromContext(ctx)
+	logger.InfoContext(ctx, fmt.Sprintf("Start job %s", jobName))
+
+	checkinId := sentry.CaptureCheckIn(
+		&sentry.CheckIn{
+			MonitorSlug: jobName,
+			Status:      sentry.CheckInStatusInProgress,
+		},
+		nil,
+	)
+
+	err = fn(ctx, uc)
+	if err != nil {
+		// Known issue where Cloud Run will sometimes fail to create the unix socket to connect to CloudSQL. In this case, we don't log the error in Sentry.
+		if strings.Contains(err.Error(), "failed to connect to `host=/cloudsql/") {
+			logger.WarnContext(ctx, "Failed to create unix socket to connect to CloudSQL. Wait for the next execution of the job.")
+			return nil
+		}
```

Comment:
You may try to log more error detail if you plan to get rid of the string matching

Reply:
Hmm well I don't know, either we get an error that matches this (and we silence it on sentry because there's nothing we can do with it), or we get something different and it's caught in the base error case. I think it's ok like this